pkg/setting/setting_secure_socks_proxy.go

```diff
@@ -42,5 +43,21 @@ func readSecureSocksDSProxySettings(iniFile *ini.File) (SecureSocksDSProxySettin
 		return s, errors.New("proxy address required")
 	}
 
+	setDefaultProxyCli(s)
+
 	return s, nil
 }
+
+// setDefaultProxyCli overrides the default proxy cli for the sdk
+//
+// Note: Not optimal changing global state, but hard to not do in this case.
+func setDefaultProxyCli(cfg SecureSocksDSProxySettings) {
+	sdkproxy.Cli = sdkproxy.NewWithCfg(&sdkproxy.ClientCfg{
```

Comment:
Only now I have realized that this `Cli` existed, please see my comment https://github.com/grafana/grafana-plugin-sdk-go/pull/716#discussion_r1266572499

We should not continue with this pattern.

Reply:
Clarified: https://github.com/grafana/grafana-plugin-sdk-go/pull/716#discussion_r1266897416